### PR TITLE
added forgot password feature

### LINKED ISSUE #154: Create the Forgot Password screen
Blocked by #151 , #152 , #153 .

- [x] Create the forgot password screen as per our Hi-Fi design

This includes the GUI screen and the controller (the `ViewModel`).

This task likely makes use of `FirebaseAuthentication` that we use for login, signup, and so forth. It is common for people to implement a way to recover passwords in Flutter, [check out this example](https://dev.to/dav4thevid/how-to-implement-forgot-password-password-reset-into-your-flutter-apps-with-firebase-auth-3jjd). You're welcome to research for a better way, too.

### PR DESCRIPTION
**Pull request by mansiPaTeLL**
@holtzmak @MummenRider @SanaKhan1997 Please review this PR. Now you have to sign up using legit email address if you want to reset your password.

### CODE REVIEW

File: app/lib/core/view_models/forgot_password_view_model.dart [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import 'package:app/ui/views/sign_in_screen.dart';
+import 'package:flutter/material.dart';
+
+class ForgotPasswordViewModel extends BaseViewModel {
+  final DialogService _dialogService = locator<DialogService>();
+  final AuthenticationService _authenticationService =
+      locator<AuthenticationService>();
+  final NavigationService _navigationService = locator<NavigationService>();
+
+  void navigateToSignInScreen() =>
```
Comment (holtzmak) [inline app/lib/core/view_models/forgot_password_view_model.dart:17]: - [x] If we're coming from the SignInScreen to this screen, we should just `navigateBackTo` and not nav and replace. 

Nav and replace is meant for parallel screens. This screen is more meant for "sitting on top"

File: app/lib/core/view_models/sign_in_view_model.dart [hunk 2]
```diff
@@ -18,6 +19,9 @@ class SignInViewModel extends BaseViewModel {
   void navigateToWelcomeScreen() =>
       _navigationService.navigateBackUntil(WelcomeScreen.route);
 
+  void navigateToForgotPasswordScreen() =>
+      _navigationService.navigateAndReplace(ForgotPasswordScreen.route);
```
Comment (holtzmak) [inline app/lib/core/view_models/sign_in_view_model.dart]: - [x] Should be `navigateTo`

**Review by holtzmak — CHANGES_REQUESTED** [review]
Minor navigation changes. Thanks!

**Comment by mansiPaTeLL** [comment]
@holtzmak @MummenRider @SanaKhan1997  
Screen Shot 2021-03-12 at 1 22 13 PM
Screen Shot 2021-03-12 at 1 22 34 PM
Screen Shot 2021-03-12 at 1 22 58 PM
Screen Shot 2021-03-12 at 1 23 13 PM
Screen Shot 2021-03-12 at 1 23 28 PM

**Review by holtzmak — APPROVED** [review]
Hi Mansi, I played around with your changes on my machine. I made a necessary navigation change to the email confirmation screen, and then added some GUI changes that were bothering me too. Sorry for piggybacking on your PR.

Everything looks good now!

**Pull request merged**